src/scripts/activate-user.ts

```diff
@@ -0,0 +1,197 @@
+import { Argv } from 'yargs';
+import { RootCommand } from '..';
+import { OktaUserService, User, UserService } from './services/user-service';
+import * as okta from '@okta/okta-sdk-nodejs';
+import { oktaManageClient } from './services/client-service';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import { flow, pipe } from 'fp-ts/lib/function';
+import * as Console from 'fp-ts/lib/Console';
+import * as O from 'fp-ts/lib/Option';
+
+/**
+ * User that can be activated.
+ */
+type ActivatableUser = User & {
+  readonly status: okta.UserStatus.DEPROVISIONED | okta.UserStatus.STAGED;
+};
+
+/**
+ * Checks to see if user exists.
+ * @param service - the service to use to get the user.
+ * @param userId - the id of the user to get.
+ * @returns a TaskEither that resolves to the user if the user exists, otherwise an error message.
+ */
+const validateUserExist = (
+  service: UserService,
+  userId: string
+): TE.TaskEither<string, User> =>
+  pipe(
+    userId,
+    service.getUser,
+    TE.chain(
+      flow(
```

Comment:
**Question:** why not use `TE.fromOption` to simplify this?

Reply:
modified in commit: e743c8258d8660e8602d291c16dd3af5b11fbc97